cirq-web/cirq_ts/README.md

```diff
@@ -0,0 +1,77 @@
+## Cirq Typescript Development
+
+*This directory contains the code and instructions for Typescript visualization development in Cirq. While necessary, this is only half of the code needed to run the `cirq-web` project. For information on how to integrate projects here with Python and the wider Cirq package, see the `cirq_web` directory.*
+
+Visualizations run on [NodeJS](https://nodejs.org/en/), and we use [npm](https://www.npmjs.com/) for package management. To start developing, clone the repository and run `npm install` within this directory to install the necessary packages and begin development. You will need to install Node and npm if you haven't already.
+
+For 3D visualizations, we rely on the [three.js](https://threejs.org/) framework. Understanding their documentation is critical to understanding 3D visualizations in Cirq. 
+
+For bundling the Typescript into Javascript that can be run in the browser, and for overall ease of development, we use [Webpack](https://webpack.js.org/).
+
+### Visualization build structure
+
+A good example on the build structure of a visualization is the Bloch sphere. Reference the `src/bloch_sphere/main.ts` file and the `CirqBlochSphere.class.ts` file to see the code. The `src/bloch_sphere/` directory should serve as a guide for how Typescript visualizations in Cirq should be  
+developed in general. In the general sense, visualizations should have:
+ - A "root" folder within the `src/` directory labeled according to the title of the visualization. All files and directories for a particular visualization will live here. In the case of the Bloch Sphere, this is `bloch_sphere/`.
+ - A `components/` directory which contains classes representing different components of the larger visualization, following typical object oriented programming techniques. In the case of the Bloch sphere, you can see that we have different classes for `Axes`, `Meridians`, `Text` etc.
+ - Any additional directories with information necessary for the visualization (fonts, images, etc.). In the case of the Bloch Sphere, we can see a `fonts/` directory which holds necessary font data. 
+ - A larger class within the visualization's "root" folder which brings the individual components of the visualization together. In the case of the Bloch Sphere, this is `CirqBlochSphere.class.ts`.
```

Comment:
I don't think its important actually, when initially toying around with everything we started it that way and it just stuck. I'm going to go through and rename things anyways, delete the `*.d.ts` files, make filenames lowercase, etc, so this will definitely change.